pkg/infrastructure/azure/image.go

```diff
@@ -0,0 +1,239 @@
+package azure
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute/v4"
+	"github.com/coreos/stream-metadata-go/arch"
+	"github.com/openshift/installer/pkg/infrastructure/clusterapi"
+	"github.com/openshift/installer/pkg/rhcos"
+	"github.com/sirupsen/logrus"
+)
+
+func (p *Provider) createImages(ctx context.Context, in clusterapi.PreProvisionInput) error {
+	session, err := in.InstallConfig.Azure.Session()
+	if err != nil {
+		return fmt.Errorf("failed to get session: %w", err)
+	}
+
+	installConfig := in.InstallConfig.Config
+	platform := installConfig.Platform.Azure
+	subscriptionID := session.Credentials.SubscriptionID
+	cloudConfiguration := session.CloudConfig
+	tokenCredential := session.TokenCreds
+	resourceGroupName := platform.ClusterResourceGroupName(in.InfraID)
+
+	// Create storage account
+	storageAccountName := fmt.Sprintf("%ssa", strings.ReplaceAll(in.InfraID, "-", ""))
+	createStorageAccountOutput, err := CreateStorageAccount(ctx, &CreateStorageAccountInput{
+		SubscriptionID:     subscriptionID,
+		ResourceGroupName:  resourceGroupName,
+		StorageAccountName: storageAccountName,
+		CloudName:          platform.CloudName,
+		Region:             platform.Region,
+		Tags:               p.Tags,
+		TokenCredential:    tokenCredential,
+		CloudConfiguration: cloudConfiguration,
+	})
+	if err != nil {
+		return err
+	}
+
+	storageAccount := createStorageAccountOutput.StorageAccount
+	storageClientFactory := createStorageAccountOutput.StorageClientFactory
+	storageAccountKeys := createStorageAccountOutput.StorageAccountKeys
+
+	containerName := "vhd"
+	blobName := fmt.Sprintf("rhcos%s.vhd", randomString(5))
+
+	stream, err := rhcos.FetchCoreOSBuild(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to get rhcos stream: %w", err)
+	}
+	archName := arch.RpmArch(string(installConfig.ControlPlane.Architecture))
+	streamArch, err := stream.GetArchitecture(archName)
+	if err != nil {
+		return fmt.Errorf("failed to get rhcos architecture: %w", err)
+	}
+
+	azureDisk := streamArch.RHELCoreOSExtensions.AzureDisk
+	imageURL := azureDisk.URL
```

Comment:
Isn't this already done the rhcos/image asset?
```suggestion
	imageURL := in.RhcosImage.ControlPlane
```

Reply:
@r4f4 yes ;-) I complicated things. This can all go away.